internal/handlers/common/filter.go

```diff
@@ -486,3 +494,66 @@ func filterFieldExprBitsAnySet(fieldValue, maskValue any) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldMod handles {field: {$mod: [divisor, remainder]}} filter.
+func filterFieldMod(fieldValue, exprValue any) (bool, error) {
+	var field, divisor, remainder int64
+
+	switch f := fieldValue.(type) {
+	case int32:
+		field = int64(f)
+	case int64:
+		field = f
+	case float64:
+		if math.IsNaN(f) || math.IsInf(f, 0) {
+			return false, nil
+		}
+		field = int64(f)
+	default:
+		return false, nil
+	}
+
+	arr := exprValue.(*types.Array)
+	if arr.Len() < 2 {
+		return false, NewErrorMsg(ErrBadValue, `malformed mod, not enough elements`)
+	}
+	if arr.Len() > 2 {
+		return false, NewErrorMsg(ErrBadValue, `malformed mod, too many elements`)
+	}
+
+	switch d := must.NotFail(arr.Get(0)).(type) {
+	case int32:
+		divisor = int64(d)
+	case int64:
+		divisor = d
+	case float64:
+		if math.IsNaN(d) || math.IsInf(d, 0) {
+			return false, NewErrorMsg(ErrBadValue, `malformed mod, divisor value is invalid :: caused by :: `+
```

Comment:
what are `::`s? I mean, do we do really need them?

Reply:
All those error messages are the same as returned by MongoDB; we want to use the same for compatibility, at least for now

Reply:
Yes. errors copied from mongodb